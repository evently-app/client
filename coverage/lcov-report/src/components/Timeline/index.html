<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/Timeline</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> src/components/Timeline
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">12.7% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>16/126</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">10.61% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>7/66</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">2.44% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/41</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">13.01% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>16/123</span>
      </div>
    </div>
    <p class="quiet">
      Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
    </p>
  </div>
  <div class='status-line low'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file low" data-value="CalendarButton.js"><a href="CalendarButton.js.html">CalendarButton.js</a></td>
	<td data-value="11.11" class="pic low"><div class="chart"><div class="cover-fill" style="width: 11%;"></div><div class="cover-empty" style="width:89%;"></div></div></td>
	<td data-value="11.11" class="pct low">11.11%</td>
	<td data-value="27" class="abs low">3/27</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="4" class="abs low">0/4</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="14" class="abs low">0/14</td>
	<td data-value="11.11" class="pct low">11.11%</td>
	<td data-value="27" class="abs low">3/27</td>
	</tr>

<tr>
	<td class="file medium" data-value="EventCardPreview.js"><a href="EventCardPreview.js.html">EventCardPreview.js</a></td>
	<td data-value="50" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 50%;"></div><div class="cover-empty" style="width:50%;"></div></div></td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="10" class="abs medium">5/10</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="8" class="abs medium">4/8</td>
	<td data-value="16.67" class="pct low">16.67%</td>
	<td data-value="6" class="abs low">1/6</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="10" class="abs medium">5/10</td>
	</tr>

<tr>
	<td class="file low" data-value="index.js"><a href="index.js.html">index.js</a></td>
	<td data-value="8.99" class="pic low"><div class="chart"><div class="cover-fill" style="width: 8%;"></div><div class="cover-empty" style="width:92%;"></div></div></td>
	<td data-value="8.99" class="pct low">8.99%</td>
	<td data-value="89" class="abs low">8/89</td>
	<td data-value="5.56" class="pct low">5.56%</td>
	<td data-value="54" class="abs low">3/54</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="21" class="abs low">0/21</td>
	<td data-value="9.3" class="pct low">9.3%</td>
	<td data-value="86" class="abs low">8/86</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Apr 14 2019 20:53:00 GMT-0400 (Eastern Daylight Time)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
<script src="../../../block-navigation.js"></script>
</body>
</html>
