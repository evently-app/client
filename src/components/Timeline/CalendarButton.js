import React, { Component } from "react";
import {
	Animated,
	StyleSheet,
	View,
	TouchableOpacity,
	Linking
} from "react-native";
import { connect } from "react-redux";
import { BlurView } from "react-native-blur";
import moment from "moment";
import RNCalendarEvents from "react-native-calendar-events";

import { colors } from "../../lib/styles";
import { Header, SubHeader, Paragraph } from "../universal/Text";
import { AddEventToCalendar } from "../../api";

class CalendarButton extends Component {
	render() {
		const {
			eventName,
			start,
			end,
			eventId,
			uid,
			onAdd = () => {}
		} = this.props;

		const inputRange = [0, 50, 110, 150];

		let formatDay = date => {
			console.log("**********", date);
			var end = moment(date).format("YYYY-MM-DDTHH:mm:ss.000Z");
			var start = moment(date).format("YYYY-MM-DDTHH:mm:ss.000Z");
			return [start, end];
		};

		let addToCalendar = (title, start, end, uid, eventId) => {
			this.setState({});
			RNCalendarEvents.saveEvent(title, {
				startDate: formatDay(start)[0],
				endDate: formatDay(start)[1]
			})
				.then(res => {
					onAdd();
					//add event to Firebase
					AddEventToCalendar(uid, eventId)
						.then(res => {})
						.catch(err => {
							console.log(err);
						});
				})
				.catch(err => {
					console.log(err);
				});
		};

		let calendarAuth = (eventName, start, end, uid, eventId) => {
			RNCalendarEvents.authorizationStatus()
				.then(res => {
					if (res == "authorized") {
						console.log("RES1: ", res);
						addToCalendar(eventName, start, end);
					} else if (res == "undetermined") {
						RNCalendarEvents.authorizeEventStore()
							.then(res => {
								addToCalendar(eventName, start, end, uid, eventId);
							})
							.catch(err => {
								console.log(err);
							});
					}
					//TODO - make alert here if no access is granted
				})
				.catch(err => {
					console.log(err);
				});

			addToCalendar(eventName, start, end, uid, eventId);
		};

		return (
			<TouchableOpacity
				activeOpacity={0.9}
				style={styles.wrapper}
				onPressIn={() => {
					//addToCalendar goes here
					calendarAuth(eventName, start, end, uid, eventId);
				}}
			>
				<BlurView blurType="xlight" style={styles.button}>
					<SubHeader style={{ color: "black" }}>Add to Calendar</SubHeader>
				</BlurView>
			</TouchableOpacity>
		);
	}
}

const styles = StyleSheet.create({
	button: {
		right: 0,
		bottom: 0,
		width: 135,
		height: 28,
		position: "absolute",
		borderRadius: 10,
		justifyContent: "center",
		alignItems: "center"
	},
	wrapper: {
		right: 10,
		bottom: 10,
		width: 135,
		height: 28,
		position: "absolute",
		borderRadius: 10,
		justifyContent: "center",
		alignItems: "center"
	}
});

const mapStateToProps = ({ user }) => {
	return {
		uid: user.uid
	};
};

export default connect(mapStateToProps)(CalendarButton);
