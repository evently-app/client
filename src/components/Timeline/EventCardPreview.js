import React, { Component } from "react";
import {
	View,
	Text,
	Image,
	Alert,
	Animated,
	StyleSheet,
	TouchableWithoutFeedback,
	TouchableOpacity
} from "react-native";
import { BlurView } from "react-native-blur";
import { Header, SubHeader } from "../universal/Text";
import CalendarButton from "./CalendarButton";
import { SCREEN_HEIGHT, SCREEN_WIDTH } from "../../lib/constants";
import LinearGradient from "react-native-linear-gradient";
import { VibrancyView } from "react-native-blur";

const GoingButton = () => {
	return (
		<TouchableOpacity
			activeOpacity={0.9}
			style={styles.outline}
		>
		<BlurView blurType="regular" style={styles.button}>
			<SubHeader style={{color: "white"}}>
				Going
			</SubHeader>
		</BlurView>
		</TouchableOpacity>
	);
};

const CARD_HEIGHT = 150;

class EventCardPreview extends Component {

	constructor(props) {
    	super(props);
    	this.state = {isAddedToCalendar: this.props.isAddedToCalendar}
    }

	componentWillMount() {
		this.scale = new Animated.Value(1);
		this.actionScale = new Animated.Value(1);
	}



	render() {

		const { imageUrl, startTime, endTime, date, title, momentStartDate, momentEndDate, uid, id, action, onPress, onAction } = this.props;

		return (
			<TouchableWithoutFeedback
				onPressIn={() => {
					Animated.timing(this.scale, {
						toValue: 1.03,
						duration: 100
					}).start();
				}}
				onPressOut={() => {
					Animated.timing(this.scale, {
						toValue: 1,
						duration: 100
					}).start();
				}}
				onPress={onPress}
			>
				<Animated.View
					style={[
						styles.wrapper,
						{
							transform: [{ scale: this.scale }]
						}
					]}
				>
					<Image
						source={{ uri: this.imageUrl }}
						resizeMode="cover"
						style={styles.image}
					/>
					<LinearGradient
						style={styles.gradient}
						locations={[0, 0.9]}
						colors={["rgba(0,0,0,0.5)", "rgba(0,0,0,0.7)"]}
					/>
					<Header style={styles.header}>{title}</Header>
					<SubHeader>
						{startTime}
						{!!endTime && ` - ${endTime}`}
					</SubHeader>
					{!!date && <SubHeader>{date}</SubHeader>}
				      {this.state.isAddedToCalendar != true ? (
				        <CalendarButton 
							eventName={title} 
							start={momentStartDate}
							end={momentEndDate}
							uid={uid}
							eventId={id}
						/> 
				      ) : (
				        <GoingButton /> 
				      )}
					<TouchableWithoutFeedback
						onPressIn={() => {
							Animated.timing(this.actionScale, {
								toValue: 1.1,
								duration: 100
							}).start();
						}}
						onPressOut={() => {
							Animated.timing(this.actionScale, {
								toValue: 1,
								duration: 100
							}).start();
						}}
						onPress={onAction}
					>
						<Animated.View
							style={[
								styles.actionWrapper,
								{
									transform: [{ scale: this.actionScale }]
								}
							]}
						>
							{!!action && (
								<VibrancyView style={styles.action} blurType="xlight">
									<Header style={styles.actionText}>{action}</Header>
								</VibrancyView>
							)}
						</Animated.View>
					</TouchableWithoutFeedback>
				</Animated.View>
			</TouchableWithoutFeedback>
		);
	}
}

const styles = StyleSheet.create({
	wrapper: {
		width: SCREEN_WIDTH - 50,
		padding: 10,
		borderRadius: 10,
		height: CARD_HEIGHT,
		backgroundColor: "rgba(255, 255, 255, 0.3)",
		overflow: "hidden",
		marginBottom: 20,
		marginHorizontal: 20
	},
	image: {
		width: SCREEN_WIDTH - 40,
		height: CARD_HEIGHT,
		position: "absolute",
		left: 0,
		right: 0
	},
	gradient: {
		width: SCREEN_WIDTH - 40,
		height: CARD_HEIGHT,
		position: "absolute",
		top: 0,
		left: 0
	},
	header: {
		marginBottom: 5
	},
	actionWrapper: {
		position: "absolute",
		bottom: 10,
		right: 10
	},
	button: {
		height: 10,
		bottom: 10,
		width: 135,
		height: 28,
		position: "absolute",
		borderRadius: 10,
		justifyContent: "center",
	 	alignItems: "center"

	},
	action: {
		paddingHorizontal: 12,
		paddingVertical: 8,
		borderRadius: 10
	},
	actionText: {
		// color: "black"
	},
	outline: {
		//backgroundColor: "rgba(255,255,255,0.5)",
		right: 10,
		bottom: 10,
		width: 135,
		height: 28,
		position: "absolute",
		borderRadius: 10,
		justifyContent: "center",
	 	alignItems: "center"

	}
});

export default EventCardPreview;
