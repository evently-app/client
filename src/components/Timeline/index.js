import React, { Component } from "react";
import {
	View,
	ScrollView,
	StyleSheet,
	SectionList,
	PanResponder,
	Text,
	Animated,
	Linking,
	Alert
} from "react-native";
import { Header, SubHeader } from "../universal/Text";
import EventCardPreview from "./EventCardPreview";
import { BlurView } from "react-native-blur";
import { connect } from "react-redux";

import { SCREEN_WIDTH, SCREEN_HEIGHT, IS_X } from "../../lib/constants";
import { LoadTimeline } from "../../redux/timeline";
import moment from "moment";

const SCROLL_BAR_HEIGHT = SCREEN_HEIGHT - (IS_X ? 180 : 140);
const SCROLL_BAR_WIDTH = 5;
const SCROLL_INDICATOR_HEIGHT = 50;

const CARD_HEIGHT = 170;
const SECTION_HEADER_HEIGHT = 70;

// function that takes date object and returns readable time
function formatAMPM(date) {
	let hours = date.getHours();
	let minutes = date.getMinutes();
	const ampm = hours >= 12 ? "pm" : "am";
	hours = hours % 12;
	hours = hours ? hours : 12;
	minutes = !!minutes ? (minutes < 10 ? `0${minutes}` : minutes) : "00";
	return hours + ":" + minutes + " " + ampm;
}

// takes date object returns readable day
function formatDay(date) {
	return moment(date).format("ddd MMMM Do");
}

function sortFn(a, b) {
	const aTime = new Date(a.startTime);
	const bTime = new Date(b.startTime);
	return aTime < bTime ? -1 : aTime > bTime ? 1 : 0;
}

function compileSections(data) {
	let Past = [];
	let Today = [];
	let Tomorrow = [];
	let Later = [];
	for (let i = 0; i < data.length; i++) {
		const item = data[i];

		console.log("ITEM: ", item)

		let now = new Date();
		let tomorrow = new Date();
		tomorrow.setDate(tomorrow.getDate() + 1);
		let startDate = new Date(item.startTime);
		let endDate = new Date(item.endTime);

		if (startDate.setHours(0, 0, 0, 0) < now.setHours(0, 0, 0, 0)) {
			Past.push(item);
		} else if (startDate.setHours(0, 0, 0, 0) == now.setHours(0, 0, 0, 0)) {
			Today.push(item);
		} else if (
			startDate.setHours(0, 0, 0, 0) == tomorrow.setHours(0, 0, 0, 0)
		) {
			Tomorrow.push(item);
		} else {
			Later.push(item);
		}
	}

	// sort sections by date
	Past.sort(sortFn);
	Today.sort(sortFn);
	Tomorrow.sort(sortFn);
	Later.sort(sortFn);

	let sections = [];
	if (Past.length > 0) {
		sections.push({ title: "Past", data: Past });
	}
	if (Today.length > 0) {
		sections.push({ title: "Today", data: Today });
	}
	if (Tomorrow.length > 0) {
		sections.push({ title: "Tomorrow", data: Tomorrow });
	}
	if (Later.length > 0) {
		sections.push({ title: "Later", data: Later });
	}

	let SECTION_LIST_HEIGHT =
		sections.length * SECTION_HEADER_HEIGHT +
		CARD_HEIGHT * data.length -
		(SCREEN_HEIGHT - (IS_X ? 40 : 20));

	if (SECTION_LIST_HEIGHT < 0) SECTION_LIST_HEIGHT = 0;

	return { sections, SECTION_LIST_HEIGHT };
}

class Timeline extends Component {
	yOffset = new Animated.Value(0);

	onScroll = Animated.event(
		[{ nativeEvent: { contentOffset: { y: this.yOffset } } }],
		{
			useNativeDriver: true
		}
	);

	shouldComponentUpdate(nextProps, nextState) {
		if (nextProps.timeline.length > this.props.timeline.length) {
			return true;
		} else {
			return false;
		}
	}

	componentWillMount() {
		this._panResponder = PanResponder.create({
			onMoveShouldSetResponderCapture: () => true,
			onMoveShouldSetPanResponderCapture: () => true,

			onPanResponderGrant: (e, gestureState) => {},

			onPanResponderMove: (e, { y0, dy }) => {
				const scrollTopDifference = IS_X ? 80 : 60;
				const scrollPosition = y0 - scrollTopDifference + dy;
				let scrollPercentage = scrollPosition / SCROLL_BAR_HEIGHT;
				scrollPercentage =
					scrollPercentage > 0
						? scrollPercentage < 1
							? scrollPercentage
							: 1
						: 0;

				const { sections, SECTION_LIST_HEIGHT } = compileSections(
					this.props.timeline
				);
				this.yOffset.setValue(scrollPercentage * SECTION_LIST_HEIGHT);
				// this.Timeline.getNode().scrollTo({
				// 	y: scrollPercentage * SECTION_LIST_HEIGHT,
				// 	animated: false
				// });
			},

			onPanResponderRelease: (e, { vx, vy }) => {}
		});

		this.props
			.LoadTimeline()
			.then(() => {
				console.log("timeline synced");
			})
			.catch(error => {
				console.log(error);
			});
	}

	render() {
		const { sections, SECTION_LIST_HEIGHT } = compileSections(
			this.props.timeline
		);

		const animatedScrollIndicator = {
			transform: [
				{
					translateY: this.yOffset.interpolate({
						inputRange: [0, SECTION_LIST_HEIGHT],
						outputRange: [0, SCROLL_BAR_HEIGHT - SCROLL_INDICATOR_HEIGHT],
						extrapolate: "clamp"
					})
				}
			]
		};

		const AnimatedSectionList = Animated.createAnimatedComponent(SectionList);
		return (
			<View style={styles.wrapper}>
				<AnimatedSectionList
					ref={Timeline => (this.Timeline = Timeline)}
					initialNumToRender={4}
					style={styles.sectionList}
					onScroll={this.onScroll}
					renderItem={({ item, index, section }) => {
						console.log("ITEM IS: ", item.id)
						let startDate = new Date(item.startTime);
						let endDate = new Date(item.endTime);
						return (
							<EventCardPreview
								key={item.id}
								title={item.eventName}
								imageUrl={item.imageUrl}
								startTime={formatAMPM(startDate)}
								endTime={formatAMPM(endDate)}
								date={
									["Past", "Later"].includes(section.title)
										? formatDay(startDate)
										: null
								}
								momentStartDate={item.startTime}
								momentEndDate={item.endTime}
								action={section.title != "Past" ? item.action : null}
								onAction={() => {
									// Alert.alert(`action for ${item.id}`);
								}}
								onPress={() => {
									if (item.ticketUrl) {
										Linking.openURL(item.ticketUrl);
									}
								}}
							/>
						);
					}}
					renderSectionHeader={({ section: { title } }) => (
						<View style={styles.sectionHeader}>
							<Header>{title}</Header>
						</View>
					)}
					sections={sections}
					keyExtractor={(item, index) => item + index}
				/>
				{SECTION_LIST_HEIGHT != 0 && (
					<View style={styles.scrollContainer}>
						<Animated.View
							{...this._panResponder.panHandlers}
							style={[styles.scrollIndicator, animatedScrollIndicator]}
						/>
					</View>
				)}
			</View>
		);
	}
}

const styles = StyleSheet.create({
	scrollContainer: {
		position: "absolute",
		right: 10,
		top: IS_X ? 80 : 60,
		width: SCROLL_BAR_WIDTH,
		height: SCROLL_BAR_HEIGHT,
		backgroundColor: "rgba(255, 255, 255, 0.12)",
		borderRadius: SCROLL_BAR_WIDTH / 2
	},
	scrollIndicator: {
		position: "absolute",
		right: 0,
		top: 0,
		width: SCROLL_BAR_WIDTH,
		height: SCROLL_INDICATOR_HEIGHT,
		backgroundColor: "lightgray",
		borderRadius: SCROLL_BAR_WIDTH / 2
	},
	sectionList: {
		overflow: "hidden",
		marginTop: IS_X ? 30 : 10
		// marginBottom: IS_X ? 90 : 70
	},
	sectionHeader: {
		width: SCREEN_WIDTH,
		paddingHorizontal: 20,
		paddingVertical: 20,
		position: "relative",
		marginBottom: 10,
		backgroundColor: "rgba(0,0,0,0.9)"
	},
	wrapper: {
		flex: 1,
		width: SCREEN_WIDTH,
		height: SCREEN_HEIGHT,
		flexDirection: "column",
		justifyContent: "flex-start",
		alignItems: "flex-start"
	}
});

const mapStateToProps = state => {
	return {
		timeline: state.timeline.timeline
	};
};

const mapDispatchToProps = {
	LoadTimeline
};

export default connect(
	mapStateToProps,
	mapDispatchToProps
)(Timeline);
