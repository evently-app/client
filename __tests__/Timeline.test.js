import "react-native";
import React from "react";
import Timeline from "../src/components/Timeline";
import renderer from "react-test-renderer";

test("Timeline renders correctly", () => {
	const tree = renderer.create(<Timeline />).toJSON();
	expect(tree).toMatchSnapshot();
});
