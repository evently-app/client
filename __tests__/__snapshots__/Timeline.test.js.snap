// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add to calendar button renders correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`Event card preview renders correctly 1`] = `
<TouchableWithoutFeedback
  onPressIn={[Function]}
  onPressOut={[Function]}
>
  <AnimatedComponent
    style={
      Array [
        Object {
          "backgroundColor": "rgba(255, 255, 255, 0.3)",
          "borderRadius": 10,
          "height": 150,
          "marginBottom": 20,
          "marginHorizontal": 20,
          "overflow": "hidden",
          "padding": 10,
          "width": 700,
        },
        Object {
          "transform": Array [
            Object {
              "scale": 1,
            },
          ],
        },
      ]
    }
  >
    <Image
      resizeMode="cover"
      source={
        Object {
          "uri": undefined,
        }
      }
      style={
        Object {
          "height": 150,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "width": 710,
        }
      }
    />
    <LinearGradient
      colors={
        Array [
          "rgba(0,0,0,0.5)",
          "rgba(0,0,0,0.7)",
        ]
      }
      end={
        Object {
          "x": 0.5,
          "y": 1,
        }
      }
      locations={
        Array [
          0,
          0.9,
        ]
      }
      start={
        Object {
          "x": 0.5,
          "y": 0,
        }
      }
      style={
        Object {
          "height": 150,
          "left": 0,
          "position": "absolute",
          "top": 0,
          "width": 710,
        }
      }
    />
    <Header
      style={
        Object {
          "marginBottom": 5,
        }
      }
    />
    <SubHeader />
    <Connect(CalendarButton) />
    <TouchableWithoutFeedback
      onPressIn={[Function]}
      onPressOut={[Function]}
    >
      <AnimatedComponent
        style={
          Array [
            Object {
              "bottom": 10,
              "position": "absolute",
              "right": 10,
            },
            Object {
              "transform": Array [
                Object {
                  "scale": 1,
                },
              ],
            },
          ]
        }
      />
    </TouchableWithoutFeedback>
  </AnimatedComponent>
</TouchableWithoutFeedback>
`;

exports[`Timeline renders correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
